src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts

```diff
@@ -0,0 +1,174 @@
+import { viewsContainersExtensionPoint } from 'vs/workbench/api/browser/viewsContainersExtensionPoint';
+
+export const DataExplorerViewlet = {
+	DataExplorer: 'dataExplorer'
+};
+export const VIEWLET_ID = 'workbench.view.dataExplorer';
+export const VIEW_CONTAINER: ViewContainer = Registry.as<IViewContainersRegistry>(ViewContainerExtensions.ViewContainersRegistry).registerViewContainer(VIEWLET_ID);
+
+
+
+interface IUserFriendlyViewDescriptor {
+	id: string;
+	name: string;
+	when?: string;
+}
+
+const viewDescriptor: IJSONSchema = {
+	type: 'object',
+	properties: {
+		id: {
+			description: localize('vscode.extension.contributes.view.id', 'Identifier of the view. Use this to register a data provider through `vscode.window.registerTreeDataProviderForView` API. Also to trigger activating your extension by registering `onView:${id}` event to `activationEvents`.'),
+			type: 'string'
+		},
+		name: {
+			description: localize('vscode.extension.contributes.view.name', 'The human-readable name of the view. Will be shown'),
+			type: 'string'
+		},
+		when: {
+			description: localize('vscode.extension.contributes.view.when', 'Condition which must be true to show this view'),
+			type: 'string'
+		},
+	}
+};
+
+const dataExplorerContribution: IJSONSchema = {
+	description: localize('extension.contributes.dataExplorer', "Contributes views to the editor"),
+	type: 'object',
+	properties: {
+		'dataExplorer': {
+			description: localize('extension.dataExplorer', "Contributes views to Data Explorer container in the Activity bar"),
+			type: 'array',
+			items: viewDescriptor,
+			default: []
+		}
+	},
+	additionalProperties: {
+		description: localize('dataExplorer.contributed', "Contributes views to contributed views container"),
+		type: 'array',
+		items: viewDescriptor,
+		default: []
+	}
+};
+
+
+const dataExplorerExtensionPoint: IExtensionPoint<{ [loc: string]: IUserFriendlyViewDescriptor[] }> = ExtensionsRegistry.registerExtensionPoint<{ [loc: string]: IUserFriendlyViewDescriptor[] }>('dataExplorer', [viewsContainersExtensionPoint], dataExplorerContribution);
+
+class DataExplorerContainerExtensionHandler implements IWorkbenchContribution {
+
+	private viewContainersRegistry: IViewContainersRegistry;
+
+	constructor(
+		@IInstantiationService private instantiationService: IInstantiationService
+	) {
+		this.viewContainersRegistry = Registry.as<IViewContainersRegistry>(ViewContainerExtensions.ViewContainersRegistry);
+		this.handleAndRegisterCustomViews();
+	}
+
+	private handleAndRegisterCustomViews() {
+		if (process.env.NODE_ENV !== 'development') {
+			return;
+		}
+		dataExplorerExtensionPoint.setHandler(extensions => {
+			for (let extension of extensions) {
+				const { value, collector } = extension;
+
+				forEach(value, entry => {
+					if (!this.isValidViewDescriptors(entry.value, collector)) {
+						return;
+					}
+
+					let container = this.getViewContainer(entry.key);
```

Comment:
I don't think you need this, this contribution point only applies to the data explorer viewlet, so we can just use the VIEWLET_ID directly

Reply:
Fixed